File: 程式設計：資料結構答案檔/Python/501.py

num_activities = int(input())

act_lst = []
for _ in range(num_activities):
    start, end = eval(input())
    act_lst.append([start, end])

act_lst.sort(key=lambda x:x[1])

select_lst = [act_lst[0]]
for i in range(1, len(act_lst)):
    if select_lst[-1][1] <= act_lst[i][0]:
        select_lst.append(act_lst[i])

print(len(select_lst))

File: 程式設計：資料結構答案檔/C#/501.cs
using System;
using System.Collections.Generic;

class Activity
{
    public int Start { get; set; }
    public int End { get; set; }

    public Activity(int start, int end)
    {
        Start = start;
        End = end;
    }
}

class Program
{
    static void Main()
    {
        int numActivities = int.Parse(Console.ReadLine());
        List<Activity> activities = new List<Activity>();

        for (int i = 0; i < numActivities; i++)
        {
            string[] input = Console.ReadLine().Split(',');
            int start = int.Parse(input[0]);
            int end = int.Parse(input[1]);
            activities.Add(new Activity(start, end));
        }

        activities.Sort((a, b) => a.End.CompareTo(b.End));

        int selectedActivities = 1;
        Activity currentActivity = activities[0];
        for (int i = 1; i < numActivities; i++)
        {
            if (currentActivity.End <= activities[i].Start)
            {
                currentActivity = activities[i];
                selectedActivities++;
            }
        }

        Console.WriteLine(selectedActivities);
    }
}
